### ISSUE TITLE
Error: fatal: not in a git directory

### ISSUE THREAD

**Issue by Intrepidd** [rickstaa/action-create-tag#10]
This action has been workign flawlessly for months, thanks !

Since yesterday I now get the `fatal: not in a git directory` error when running the action. The action version has not changed so this may be an issue with github actions itself or something may have changed ?

```
Run rickstaa/action-create-tag@v1
  with:
    tag: latest_release
    force_push_tag: true
    commit_sha: xxxx
    github_token: ***
/usr/bin/docker run --name xxx --label xxx --workdir /github/workspace --rm -e INPUT_TAG -e INPUT_FORCE_PUSH_TAG -e INPUT_COMMIT_SHA -e INPUT_GITHUB_TOKEN -e INPUT_MESSAGE -e HOME -e GITHUB_JOB -e GITHUB_REF -e GITHUB_SHA -e GITHUB_REPOSITORY -e GITHUB_REPOSITORY_OWNER -e GITHUB_RUN_ID -e GITHUB_RUN_NUMBER -e GITHUB_RETENTION_DAYS -e GITHUB_RUN_ATTEMPT -e GITHUB_ACTOR -e GITHUB_WORKFLOW -e GITHUB_HEAD_REF -e GITHUB_BASE_REF -e GITHUB_EVENT_NAME -e GITHUB_SERVER_URL -e GITHUB_API_URL -e GITHUB_GRAPHQL_URL -e GITHUB_REF_NAME -e GITHUB_REF_PROTECTED -e GITHUB_REF_TYPE -e GITHUB_WORKSPACE -e GITHUB_ACTION -e GITHUB_EVENT_PATH -e GITHUB_ACTION_REPOSITORY -e GITHUB_ACTION_REF -e GITHUB_PATH -e GITHUB_ENV -e GITHUB_STEP_SUMMARY -e RUNNER_OS -e RUNNER_ARCH -e RUNNER_NAME -e RUNNER_TOOL_CACHE -e RUNNER_TEMP -e RUNNER_WORKSPACE -e ACTIONS_RUNTIME_URL -e ACTIONS_RUNTIME_TOKEN -e ACTIONS_CACHE_URL -e GITHUB_ACTIONS=true -e CI=true -v "/var/run/docker.sock":"/var/run/docker.sock" -v "/home/runner/work/_temp/_github_home":"/github/home" -v "/home/runner/work/_temp/_github_workflow":"/github/workflow" -v "/home/runner/work/_temp/_runner_file_commands":"/github/file_commands" -v "/home/runner/work/xxx":"/github/workspace" xxx:xxx
fatal: not in a git directory
```

**Comment by AgiMaulana-Justika** [comment]
it's happen to me too

**Comment by rickstaa** [comment]
@Intrepidd Thanks for reporting this. This might be due to spec changes in Git v2.35.2. Other actions I am a collaborator of also have problems. We might be able to use the workaround in https://github.com/reviewdog/action-yamllint/pull/19. I will try to investigate this at the weekend.

**Comment by Intrepidd** [comment]
Fabulous, thanks a lot

**Comment by Intrepidd** [comment]
I opened #11 if you want to take a look into it :)

**Comment by bakoontz2** [comment]
> I opened #11 if you want to take a look into it :)

I forked and tried your fix, unfortunately still getting the same error.

**Comment by Kobzol** [comment]
I have also tried this fix manually in my CI (as a step right before running `action-create-tag`) and it didn't help :/

**Comment by tiagoalmeidadarosa** [comment]
@bakoontz2 and @Kobzol In my fork I did a change in the docker file too (I just updated the alpine version to the latest) after adding the change in the #11 and it worked, look here: https://github.com/tiagoalmeidadarosa/action-create-tag/commit/6bf19be5aa9018f7d72463be24d1b66b546f9d10

**Comment by Kobzol** [comment]
Right, using a different git version is also probably a part of the cure. For me, CI broke when the Docker image started using `2.34.2-r0` instead of `2.34.1-r0`. Maybe it would be useful to pin the git version in the Dockerfile.

**Comment by bakoontz2** [comment]
@tiagoalmeidadarosa I'll take a look. 

I followed the trail [here](https://github.com/reviewdog/reviewdog/issues/1158) and ended up hacking my build yaml:

```
    - uses: actions/checkout@v2

    - name: "Change perms on GITHUB_WORKSPACE"
      run: |
        sudo chown -R root:root $GITHUB_WORKSPACE
        sudo chmod -R 0777 $GITHUB_WORKSPACE
```

But I certainly don't consider this a real fix.

**Comment by rickstaa** [comment]
I quickly looked at the issue, and it seems to be caused by https://github.blog/2022-04-12-git-security-vulnerability-announced/. People came up with multiple workarounds in the last few days to fix this issue ([this StackOverflow post](https://stackoverflow.com/questions/71901632/fatal-unsafe-repository-home-repon-is-owned-by-someone-else). Two of these workarounds are found in #11, and https://github.com/rickstaa/action-create-tag/issues/10#issuecomment-1103134487 are two. 

A statement from the GitHub theme is found at https://github.com/actions/checkout/issues/766. They already incorporated a fix for this in the GitHub/checkout action and are now looking for a fix at the actions ecosystem level. For now I think the best solution is to use @Kobzol suggestion and temporary fix the git version to `2.34.1-r0`. 

I'm also happy to incorporate the workarounds given #11. Let me know what you think.

**Comment by bakoontz2** [comment]
@rickstaa #11 didn't work for me, so I'm not sure what that will accomplish.

**Comment by bakoontz2** [comment]
@rickstaa Working on that fix now...

**Comment by rickstaa** [comment]
Great! If you want to test a solution you can use https://github.com/rickstaa/action-test-repo/tree/test/action-create-tag for testing PR found in this repo (see https://github.com/rickstaa/action-test-repo/pull/67).

**Comment by rickstaa** [comment]
@bakoontz2 I tried pinning the GitHub version to fix this issue (see #12). I, however, was not able to create a working docker in the short amount of time I had available for fixing #10. I tested the @Intrepidd solution, and it seems to work on all my systems. For now, I will therefore merge #10 into the main branch. It is both the recommended workaround (see https://github.com/actions/checkout/issues/766) and the solution used in the majority of actions affected by this bug.

Can you please provide a reproducible example of why #10 is not working for you?

**Comment by rickstaa** [comment]
I applied #11 to the main branch please test it out. 🚀

**Comment by rickstaa** [comment]
@bakoontz2 and @Kobzol thanks a lot for letting me know! The fix should also work with `checkout@v2`. I added `checkout@v3` to the README just to be sure people do not get this error for other steps in the same job (see https://github.com/actions/checkout/issues/766).
